arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -386,89 +386,83 @@ static inline bool isStopping() {
   return application_features::ApplicationServer::isStopping();
 }
 
-arangodb::Result SynchronizeShard::getReadLock(std::string const& endpoint,
-                                               std::string const& database,
-                                               std::string const& collection,
-                                               std::string const& clientId, uint64_t rlid,
-                                               bool soft, double timeout) {
+arangodb::Result SynchronizeShard::getReadLock(
+  std::string const& endpoint, std::string const& database,
+  std::string const& collection, std::string const& clientId,
+  uint64_t rlid, bool soft, double timeout) {
+
+  // This function can be implemented in a more robust manner for server
+  // versions > 3.3. Starting with 3.4 the POST requests to the read lock API
```

Comment:
Actually, it can only be implemented more robustly in versions > 3.4, since 3.4 needs to be able to upgrade from 3.3 and thus temporarily still work with 3.3 servers.